cdap-app-templates/cdap-etl/cdap-etl-api-spark/src/main/java/co/cask/cdap/etl/api/streaming/StreamingSource.java

```diff
@@ -47,4 +51,23 @@
   public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
     // no-op
   }
+
+  /**
+   * Invoke this method in {@code getStream} method to register the lineage.
+   *
+   * @param referenceName reference name of the source
+   * @param streamingContext {@link StreamingContext} object
+   * @throws TransactionFailureException thrown if there was an error in registering lineage
+   * @throws DatasetManagementException thrown if there was an error in creating the reference external dataset
+   */
+  protected void registerUsage(final String referenceName, StreamingContext streamingContext)
```

Comment:
don't think we should have this method if we have one in StreamingContext. Should move the getDataset() call into the registerUsage() method in StreamingContext